packet/client.go

```diff
@@ -10,25 +9,22 @@ import (
 	"os"
 	"runtime"
 
-	"github.com/packethost/cacher/client"
-	"github.com/packethost/cacher/protos/cacher"
+	cacherClient "github.com/packethost/cacher/client"
 	"github.com/packethost/pkg/env"
 	"github.com/pkg/errors"
 	"github.com/tinkerbell/boots/httplog"
-	"google.golang.org/grpc"
+	tinkClient "github.com/tinkerbell/tink/client"
 )
 
 type hardwareGetter interface {
-	ByMAC(context.Context, *cacher.GetRequest, ...grpc.CallOption) (*cacher.Hardware, error)
-	ByIP(context.Context, *cacher.GetRequest, ...grpc.CallOption) (*cacher.Hardware, error)
 }
 
 type Client struct {
-	http          *http.Client
-	baseURL       *url.URL
-	consumerToken string
-	authToken     string
-	cacher        hardwareGetter
+	http           *http.Client
+	baseURL        *url.URL
+	consumerToken  string
+	authToken      string
+	hardwareClient hardwareGetter
 }
 
 func NewClient(consumerToken, authToken string, baseURL *url.URL) (*Client, error) {
```

Comment:
Is this for DHCP client? wondering why we need client functions here in boots

Reply:
The `hardwareClient` refers to the tink/cacher clients responsible for getting the hardware data, see the usage here: https://github.com/tinkerbell/boots/blob/8ecd0d6d08f11d4d6e4bdb9c6b4893815e060ac9/packet/client.go#L46

Reply:
as discussed move to workflow